packages/client/components/DashboardAvatars/DashboardAvatars.tsx

```diff
@@ -64,93 +54,107 @@ const StyledButton = styled(PlainButton)({
+    return connectedUsers.concat(offlineUsers)
   }, [teamMembers])
   const {submitting, onError, onCompleted, submitMutation} = useMutationProps()
 
-  const handleClick = (clickedOverflow: boolean) => {
-    if (!submitting) {
-      submitMutation()
-      ToggleTeamDrawerMutation(
-        atmosphere,
-        {teamId, teamDrawerType: 'manageTeam'},
-        {onError, onCompleted}
-      )
-      commitLocalUpdate(atmosphere, (store) => {
-        const viewer = store.getRoot().getLinkedRecord('viewer')
-        const teamMember = viewer?.getLinkedRecord('teamMember', {teamId})
-        const memberInFocus = teamMembers[clickedOverflow ? maxAvatars - 1 : 0]
-        if (!teamMember || !memberInFocus) return
-        const {id: teamMemberId} = memberInFocus
-        teamMember.setValue(teamMemberId, 'manageTeamMemberId')
-      })
-    }
+  const handleClick = (userId?: string) => {
+    if (submitting) return
+    submitMutation()
+    ToggleTeamDrawerMutation(
+      atmosphere,
+      {teamId, teamDrawerType: 'manageTeam'},
+      {onError, onCompleted}
+    )
+    if (!userId) return
+    commitLocalUpdate(atmosphere, (store) => {
+      const viewer = store.getRoot().getLinkedRecord('viewer')
+      const teamMember = viewer?.getLinkedRecord('teamMember', {teamId})
+      if (!teamMember) return
+      const memberInFocus = teamMembers.find((teamMember) => teamMember.user.id === userId)
+      if (!memberInFocus) return
+      const {id: teamMemberId} = memberInFocus
+      teamMember.setValue(teamMemberId, 'manageTeamMemberId')
+    })
   }
 
   return (
-    <AvatarsList isDesktop={isDesktop}>
-      <AvatarsWrapper>
-        {sortedAvatars.map((teamMember) => {
-          return (
-            <ErrorBoundary key={`dbAvatar${teamMember.id}`}>
-              <DashboardAvatar teamMember={teamMember} />
-            </ErrorBoundary>
-          )
-        })}
-        {overflowCount > 0 && (
-          <OverflowWrapper onClick={() => handleClick(true)}>
-            <OverflowCount>{`+${overflowCount}`}</OverflowCount>
-          </OverflowWrapper>
-        )}
-      </AvatarsWrapper>
-      <StyledButton onClick={() => handleClick(false)}>Manage Team</StyledButton>
-    </AvatarsList>
+    <Row ref={rowRef}>
+      <Column>
+        <AvatarsWrapper totalUsers={sortedUsers.length}>
+          <AvatarList
+            users={sortedUsers}
+            containerRef={rowRef}
+            size={ElementWidth.DASHBOARD_AVATAR}
+            borderColor={PALETTE.SLATE_200}
+            onOverflowClick={() => handleClick()}
+            onUserClick={(userId) => handleClick(userId)}
+          />
+        </AvatarsWrapper>
+        <StyledButton onClick={() => handleClick()}>Manage Team</StyledButton>
+      </Column>
+    </Row>
   )
 }
 
+graphql`
+  fragment DashboardAvatars_teamMember on TeamMember {
+    ...TeamMemberAvatarMenu_teamMember
+    ...LeaveTeamModal_teamMember
```

Comment:
-1 this fragment is repeating a lot of the logic below. Which fields do we need here?